lib/modules/manager/pep621/processors/pdm.ts

```diff
@@ -64,9 +65,12 @@ export class PdmProcessor implements PyProjectProcessor {
         logger.debug('No pdm.lock found');
         return null;
       }
+
+      const def = parsePyProject(packageFileName, newPackageFileContent);
```

Comment:
we should avoid to parse the file more that once while extract or artifacts update